src/LanguageServer/Impl/Diagnostics/DiagnosticsService.cs

```diff
@@ -126,9 +138,11 @@ private sealed class DocumentDiagnostics {
                     _severityMap = value;
                     foreach (var d in _diagnostics) {
                         d.Value.Changed = true;
-                        _lastChangeTime = DateTime.Now;
                     }
+
                     PublishDiagnostics();
+
+                    _lastChangeTime = DateTime.Now;
```

Comment:
_lastChangeTime is read onIdle. all write is under lock, but on idle read it without lock. DateTime''s only field is UInt64 so probably okay for 64bit machine, but in 32bit machine might not thread safe? do we care?